testsuite/src/main/java/io/netty/testsuite/transport/socket/SocketConnectionAttemptTest.java

```diff
@@ -60,6 +64,39 @@ public void testConnectTimeout(Bootstrap cb) throws Throwable {
         }
     }
 
+    @Test(timeout = 30000)
+    public void testConnectRefused() throws Throwable {
+        run();
+    }
+
+    public void testConnectRefused(Bootstrap cb) throws Throwable {
+        testConnectRefused0(cb, false);
+    }
+
+    @Test(timeout = 30000)
+    public void testConnectRefusedHalfClosure() throws Throwable {
+        run();
+    }
+
+    public void testConnectRefusedHalfClosure(Bootstrap cb) throws Throwable {
+        testConnectRefused0(cb, true);
+    }
+
+    private static void testConnectRefused0(Bootstrap cb, boolean halfClosure) throws Throwable {
+        ChannelHandler handler = new ChannelInboundHandlerAdapter() {
+            @Override
+            public void channelActive(ChannelHandlerContext ctx) throws Exception {
+                Channel channel = ctx.channel();
+                fail("Should have never been called: " + channel);
```

Comment:
The test method will not get this failure. You could create a new `Promise` from the `GlobalEventExecutor` and set the failure here:

    promise.setFailure(new AssertionError("should have never been called ...."));